02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,588 @@
+
+Restart the development server to load the environment variables from `.env.local`. Press the combination `Ctrl/CMD + C` on your terminal to stop the development server, then restart it by running:
+
+```bash
+yarn dev
+```
+
+You will see a message on the terminal showing that environment variables have been loaded from `.env.local`.
+
+Now you have set up the environment variables for your Next.js blog, you will create a helper function that you will use to make GraphQL queries to Squidex.
+
+Create a `lib` folder in `squidex-blog` and in this `lib` folder, create a `squidex.js` file. Add the following contents to the `squidex-blog/lib/squidex.js` file:
+
+```js
+export default async function fetchAPI(query, { variables } = {}) {
+  const res = await fetch(process.env.SQUIDEX_API_URL, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      query,
+      variables,
+    }),
+  });
+  const json = await res.json();
+
+  if (json.errors) {
+    console.error(json.errors);
+    throw new Error("Failed to fetch API");
+  }
+
+  return json.data;
+}
+```
+
+This exports a `fetchAPI` function that receives a GraphQL query and its variables as parameters, queries the Squidex API, and returns the result.
+
+You are now ready to fetch content from Squidex and display it in your Next.js blog.
+
+## Step 6: Displaying blog posts on home page
+
+In this step, you will fetch the titles of and links to your blog post and display them on the homepage of your blog.
+
+Replace the contents of `squidex-blog/pages/index.js` with this:
+
+```jsx
+import Head from "next/head";
+
+export default function Home() {
+  return (
+    <div>
+      <Head>
+        <title>My blog</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+    </div>
+  );
+}
+```
+
+Open `http://localhost:3000` in your browser if you haven't already to continue previewing changes.
+
+At the top of `pages/index.js`, import the `fetchAPI` utility from `squidex-app/lib/squidex`:
+
+```js
+import fetchAPI from "../lib/squidex";
+```
+
+Also in `index.js`, export a `getStaticProps` function defined as follows:
+
+```js
+export async function getStaticProps() {
+  const data = await fetchAPI(`
+  {
+    queryPostsContents {
+      id,
+      flatData {
+        title,
+        slug,
+      }
+    }
+  }
+  `);
+  return {
+    props: {
+      posts: data.queryPostsContents,
+    },
+  };
+}
+```
+
+This sends a GraphQL query to your Squidex app to get the slug and titles of all your blog posts and it passes this data to your page.
+
+In Next.js, `getStaticProps` is a function that is run at build time to fetch and store data needed to render a page. To learn more about fetching data in Next.js see [Data Fetching in Next.js](https://nextjs.org/docs/basic-features/data-fetching). Save the file and refresh your browser to fetch page data from the CMS.
```

Comment:
I had to read the docs for getStaticProps to understand it. The link should point to the docs directly.